[2025-10-18T16:11:31.725+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: test_hdfs_spark.hdfs_spark_test manual__2025-10-18T16:11:24.112286+00:00 [queued]>
[2025-10-18T16:11:31.765+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: test_hdfs_spark.hdfs_spark_test manual__2025-10-18T16:11:24.112286+00:00 [queued]>
[2025-10-18T16:11:31.766+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-10-18T16:11:31.832+0000] {taskinstance.py:2191} INFO - Executing <Task(SparkSubmitOperator): hdfs_spark_test> on 2025-10-18 16:11:24.112286+00:00
[2025-10-18T16:11:31.847+0000] {standard_task_runner.py:60} INFO - Started process 3651 to run task
[2025-10-18T16:11:31.859+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'test_hdfs_spark', 'hdfs_spark_test', 'manual__2025-10-18T16:11:24.112286+00:00', '--job-id', '98', '--raw', '--subdir', 'DAGS_FOLDER/test_hdfs_spark.py', '--cfg-path', '/tmp/tmp3_r06vdu']
[2025-10-18T16:11:31.869+0000] {standard_task_runner.py:88} INFO - Job 98: Subtask hdfs_spark_test
[2025-10-18T16:11:32.108+0000] {task_command.py:423} INFO - Running <TaskInstance: test_hdfs_spark.hdfs_spark_test manual__2025-10-18T16:11:24.112286+00:00 [running]> on host 2ef039423968
[2025-10-18T16:11:32.282+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='test_hdfs_spark' AIRFLOW_CTX_TASK_ID='hdfs_spark_test' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T16:11:24.112286+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-18T16:11:24.112286+00:00'
[2025-10-18T16:11:32.305+0000] {base.py:83} INFO - Using connection ID 'my_spark_conn' for task execution.
[2025-10-18T16:11:32.308+0000] {spark_submit.py:341} INFO - Spark-Submit cmd: spark-submit --master local --name arrow-spark --verbose /opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:11:46.156+0000] {spark_submit.py:492} INFO - Using properties file: null
[2025-10-18T16:11:47.686+0000] {spark_submit.py:492} INFO - Parsed arguments:
[2025-10-18T16:11:47.749+0000] {spark_submit.py:492} INFO - master                  local
[2025-10-18T16:11:47.762+0000] {spark_submit.py:492} INFO - remote                  null
[2025-10-18T16:11:47.763+0000] {spark_submit.py:492} INFO - deployMode              null
[2025-10-18T16:11:47.764+0000] {spark_submit.py:492} INFO - executorMemory          null
[2025-10-18T16:11:47.765+0000] {spark_submit.py:492} INFO - executorCores           null
[2025-10-18T16:11:47.767+0000] {spark_submit.py:492} INFO - totalExecutorCores      null
[2025-10-18T16:11:47.768+0000] {spark_submit.py:492} INFO - propertiesFile          null
[2025-10-18T16:11:47.771+0000] {spark_submit.py:492} INFO - driverMemory            null
[2025-10-18T16:11:47.778+0000] {spark_submit.py:492} INFO - driverCores             null
[2025-10-18T16:11:47.780+0000] {spark_submit.py:492} INFO - driverExtraClassPath    null
[2025-10-18T16:11:47.781+0000] {spark_submit.py:492} INFO - driverExtraLibraryPath  null
[2025-10-18T16:11:47.849+0000] {spark_submit.py:492} INFO - driverExtraJavaOptions  null
[2025-10-18T16:11:47.858+0000] {spark_submit.py:492} INFO - supervise               false
[2025-10-18T16:11:47.859+0000] {spark_submit.py:492} INFO - queue                   null
[2025-10-18T16:11:47.860+0000] {spark_submit.py:492} INFO - numExecutors            null
[2025-10-18T16:11:47.861+0000] {spark_submit.py:492} INFO - files                   null
[2025-10-18T16:11:47.867+0000] {spark_submit.py:492} INFO - pyFiles                 null
[2025-10-18T16:11:47.872+0000] {spark_submit.py:492} INFO - archives                null
[2025-10-18T16:11:47.874+0000] {spark_submit.py:492} INFO - mainClass               null
[2025-10-18T16:11:47.875+0000] {spark_submit.py:492} INFO - primaryResource         file:/opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:11:47.949+0000] {spark_submit.py:492} INFO - name                    arrow-spark
[2025-10-18T16:11:47.950+0000] {spark_submit.py:492} INFO - childArgs               []
[2025-10-18T16:11:47.951+0000] {spark_submit.py:492} INFO - jars                    null
[2025-10-18T16:11:47.953+0000] {spark_submit.py:492} INFO - packages                null
[2025-10-18T16:11:47.954+0000] {spark_submit.py:492} INFO - packagesExclusions      null
[2025-10-18T16:11:47.955+0000] {spark_submit.py:492} INFO - repositories            null
[2025-10-18T16:11:47.956+0000] {spark_submit.py:492} INFO - verbose                 true
[2025-10-18T16:11:47.958+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:47.973+0000] {spark_submit.py:492} INFO - Spark properties used, including those specified through
[2025-10-18T16:11:47.981+0000] {spark_submit.py:492} INFO - --conf and those from the properties file null:
[2025-10-18T16:11:47.982+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:47.984+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:47.985+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:51.970+0000] {spark_submit.py:492} INFO - Main class:
[2025-10-18T16:11:51.975+0000] {spark_submit.py:492} INFO - org.apache.spark.deploy.PythonRunner
[2025-10-18T16:11:52.062+0000] {spark_submit.py:492} INFO - Arguments:
[2025-10-18T16:11:52.065+0000] {spark_submit.py:492} INFO - file:/opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:11:52.066+0000] {spark_submit.py:492} INFO - null
[2025-10-18T16:11:52.067+0000] {spark_submit.py:492} INFO - Spark config:
[2025-10-18T16:11:52.068+0000] {spark_submit.py:492} INFO - (spark.app.name,arrow-spark)
[2025-10-18T16:11:52.069+0000] {spark_submit.py:492} INFO - (spark.app.submitTime,1760803911795)
[2025-10-18T16:11:52.070+0000] {spark_submit.py:492} INFO - (spark.master,local)
[2025-10-18T16:11:52.071+0000] {spark_submit.py:492} INFO - (spark.submit.deployMode,client)
[2025-10-18T16:11:52.072+0000] {spark_submit.py:492} INFO - (spark.submit.pyFiles,)
[2025-10-18T16:11:52.074+0000] {spark_submit.py:492} INFO - Classpath elements:
[2025-10-18T16:11:52.150+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:52.152+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:52.158+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:11:56.590+0000] {spark_submit.py:492} INFO - 🚀 Starting Spark HDFS Test...
[2025-10-18T16:11:56.591+0000] {spark_submit.py:492} INFO - Connecting to HDFS at hdfs://192.168.1.10:9000 ...
[2025-10-18T16:11:57.357+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:57 INFO SparkContext: Running Spark version 3.5.1
[2025-10-18T16:11:57.387+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:57 INFO SparkContext: OS info Linux, 6.6.87.2-microsoft-standard-WSL2, amd64
[2025-10-18T16:11:57.449+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:57 INFO SparkContext: Java version 17.0.16
[2025-10-18T16:11:58.065+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:58 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-10-18T16:11:59.188+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceUtils: ==============================================================
[2025-10-18T16:11:59.191+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-10-18T16:11:59.197+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceUtils: ==============================================================
[2025-10-18T16:11:59.263+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO SparkContext: Submitted application: HDFS-Test
[2025-10-18T16:11:59.453+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-10-18T16:11:59.475+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceProfile: Limiting resource is cpu
[2025-10-18T16:11:59.478+0000] {spark_submit.py:492} INFO - 25/10/18 16:11:59 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-10-18T16:12:00.161+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:00 INFO SecurityManager: Changing view acls to: ***
[2025-10-18T16:12:00.167+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:00 INFO SecurityManager: Changing modify acls to: ***
[2025-10-18T16:12:00.168+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:00 INFO SecurityManager: Changing view acls groups to:
[2025-10-18T16:12:00.172+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:00 INFO SecurityManager: Changing modify acls groups to:
[2025-10-18T16:12:00.173+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:00 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-10-18T16:12:02.566+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:02 INFO Utils: Successfully started service 'sparkDriver' on port 35281.
[2025-10-18T16:12:02.690+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:02 INFO SparkEnv: Registering MapOutputTracker
[2025-10-18T16:12:02.982+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:02 INFO SparkEnv: Registering BlockManagerMaster
[2025-10-18T16:12:03.189+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-10-18T16:12:03.251+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-10-18T16:12:03.264+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-10-18T16:12:03.551+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-897b5de7-964a-4191-8b7f-bbe196604d2d
[2025-10-18T16:12:03.674+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO MemoryStore: MemoryStore started with capacity 413.9 MiB
[2025-10-18T16:12:03.796+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:03 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-10-18T16:12:04.915+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:04 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-10-18T16:12:05.508+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:05 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-10-18T16:12:06.294+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Executor: Starting executor ID driver on host 2ef039423968
[2025-10-18T16:12:06.314+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Executor: OS info Linux, 6.6.87.2-microsoft-standard-WSL2, amd64
[2025-10-18T16:12:06.385+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Executor: Java version 17.0.16
[2025-10-18T16:12:06.412+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-10-18T16:12:06.417+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@1c590b88 for default.
[2025-10-18T16:12:06.515+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 39185.
[2025-10-18T16:12:06.516+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO NettyBlockTransferService: Server created on 2ef039423968:39185
[2025-10-18T16:12:06.520+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-10-18T16:12:06.543+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2ef039423968, 39185, None)
[2025-10-18T16:12:06.551+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO BlockManagerMasterEndpoint: Registering block manager 2ef039423968:39185 with 413.9 MiB RAM, BlockManagerId(driver, 2ef039423968, 39185, None)
[2025-10-18T16:12:06.560+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2ef039423968, 39185, None)
[2025-10-18T16:12:06.565+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:06 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2ef039423968, 39185, None)
[2025-10-18T16:12:08.768+0000] {spark_submit.py:492} INFO - 📂 Trying to read from: hdfs://192.168.1.10:9000/user/winardi/data/olist/olist_customers_dataset.csv
[2025-10-18T16:12:09.170+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:09 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-10-18T16:12:09.349+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:09 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-10-18T16:12:17.718+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:17 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: hdfs://192.168.1.10:9000/user/winardi/data/olist/olist_customers_dataset.csv.
[2025-10-18T16:12:17.722+0000] {spark_submit.py:492} INFO - java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:12:17.723+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:12:17.726+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:12:17.727+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:12:17.738+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:12:17.740+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:12:17.743+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:12:17.745+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:12:17.746+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:12:17.748+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:12:17.749+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:12:17.750+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:12:17.751+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:12:17.752+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:12:17.753+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:12:17.754+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:12:17.755+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:12:17.756+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:12:17.757+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:12:17.759+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:12:17.788+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:12:17.789+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:12:17.790+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:12:17.791+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:12:17.792+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:12:17.793+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:12:17.794+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:12:17.794+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:12:17.796+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:12:17.797+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:12:17.798+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.isDirectory(FileSystem.java:1777)
[2025-10-18T16:12:17.798+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:54)
[2025-10-18T16:12:17.799+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2025-10-18T16:12:17.800+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2025-10-18T16:12:17.801+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2025-10-18T16:12:17.801+0000] {spark_submit.py:492} INFO - at scala.Option.getOrElse(Option.scala:189)
[2025-10-18T16:12:17.803+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2025-10-18T16:12:17.803+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.csv(DataFrameReader.scala:538)
[2025-10-18T16:12:17.804+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:12:17.805+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:12:17.805+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:12:17.806+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:12:17.806+0000] {spark_submit.py:492} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-10-18T16:12:17.807+0000] {spark_submit.py:492} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-10-18T16:12:17.808+0000] {spark_submit.py:492} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2025-10-18T16:12:17.808+0000] {spark_submit.py:492} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-10-18T16:12:17.809+0000] {spark_submit.py:492} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-10-18T16:12:17.809+0000] {spark_submit.py:492} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-10-18T16:12:17.810+0000] {spark_submit.py:492} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-10-18T16:12:17.811+0000] {spark_submit.py:492} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-10-18T16:12:17.812+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:12:17.813+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:12:17.813+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:12:17.814+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:12:17.814+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:12:17.815+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:12:17.816+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:12:17.816+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:12:17.817+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:12:17.818+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:12:17.818+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:12:17.819+0000] {spark_submit.py:492} INFO - ... 40 more
[2025-10-18T16:12:20.162+0000] {spark_submit.py:492} INFO - ❌ ERROR while reading from HDFS: An error occurred while calling o29.csv.
[2025-10-18T16:12:20.165+0000] {spark_submit.py:492} INFO - : java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:12:20.170+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:12:20.196+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:12:20.200+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:12:20.211+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:12:20.213+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:12:20.231+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:12:20.236+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:12:20.239+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:12:20.243+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:12:20.244+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:12:20.289+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:12:20.290+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:12:20.291+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:12:20.292+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:12:20.293+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:12:20.294+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:12:20.296+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:12:20.297+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:12:20.298+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:12:20.300+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:12:20.301+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:12:20.304+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:12:20.306+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:12:20.307+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:12:20.308+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:12:20.309+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:12:20.310+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:12:20.311+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:12:20.312+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:12:20.313+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.exists(FileSystem.java:1760)
[2025-10-18T16:12:20.315+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4(DataSource.scala:756)
[2025-10-18T16:12:20.317+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4$adapted(DataSource.scala:754)
[2025-10-18T16:12:20.321+0000] {spark_submit.py:492} INFO - at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:384)
[2025-10-18T16:12:20.322+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-10-18T16:12:20.389+0000] {spark_submit.py:492} INFO - at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-10-18T16:12:20.395+0000] {spark_submit.py:492} INFO - at scala.util.Success.map(Try.scala:213)
[2025-10-18T16:12:20.404+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-10-18T16:12:20.405+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-10-18T16:12:20.407+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-10-18T16:12:20.408+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-10-18T16:12:20.410+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[2025-10-18T16:12:20.411+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[2025-10-18T16:12:20.488+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[2025-10-18T16:12:20.490+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[2025-10-18T16:12:20.496+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[2025-10-18T16:12:20.497+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[2025-10-18T16:12:20.498+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:12:20.499+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:12:20.500+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:12:20.502+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:12:20.503+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:12:20.504+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:12:20.504+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:12:20.505+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:12:20.506+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:12:20.508+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:12:20.509+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:12:20.510+0000] {spark_submit.py:492} INFO - ... 37 more
[2025-10-18T16:12:20.511+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:12:20.512+0000] {spark_submit.py:492} INFO - Traceback (most recent call last):
[2025-10-18T16:12:20.513+0000] {spark_submit.py:492} INFO - File "/opt/***/dags/hdfs_spark_test.py", line 29, in main
[2025-10-18T16:12:20.514+0000] {spark_submit.py:492} INFO - df = spark.read.csv(hdfs_path, header=True, inferSchema=True)
[2025-10-18T16:12:20.515+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 740, in csv
[2025-10-18T16:12:20.516+0000] {spark_submit.py:492} INFO - return self._df(self._jreader.csv(self._spark._sc._jvm.PythonUtils.toSeq(path)))
[2025-10-18T16:12:20.518+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-10-18T16:12:20.589+0000] {spark_submit.py:492} INFO - return_value = get_return_value(
[2025-10-18T16:12:20.594+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2025-10-18T16:12:20.596+0000] {spark_submit.py:492} INFO - return f(*a, **kw)
[2025-10-18T16:12:20.597+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-10-18T16:12:20.598+0000] {spark_submit.py:492} INFO - raise Py4JJavaError(
[2025-10-18T16:12:20.599+0000] {spark_submit.py:492} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o29.csv.
[2025-10-18T16:12:20.600+0000] {spark_submit.py:492} INFO - : java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:12:20.601+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:12:20.603+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:12:20.604+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:12:20.605+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:12:20.606+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:12:20.607+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:12:20.608+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:12:20.609+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:12:20.610+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:12:20.611+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:12:20.613+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:12:20.614+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:12:20.615+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:12:20.616+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:12:20.622+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:12:20.690+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:12:20.691+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:12:20.692+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:12:20.695+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:12:20.698+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:12:20.699+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:12:20.699+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:12:20.700+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:12:20.700+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:12:20.701+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:12:20.706+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:12:20.706+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:12:20.707+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:12:20.707+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:12:20.708+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.exists(FileSystem.java:1760)
[2025-10-18T16:12:20.709+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4(DataSource.scala:756)
[2025-10-18T16:12:20.710+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4$adapted(DataSource.scala:754)
[2025-10-18T16:12:20.710+0000] {spark_submit.py:492} INFO - at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:384)
[2025-10-18T16:12:20.711+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-10-18T16:12:20.711+0000] {spark_submit.py:492} INFO - at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-10-18T16:12:20.713+0000] {spark_submit.py:492} INFO - at scala.util.Success.map(Try.scala:213)
[2025-10-18T16:12:20.714+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-10-18T16:12:20.714+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-10-18T16:12:20.715+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-10-18T16:12:20.716+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-10-18T16:12:20.716+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[2025-10-18T16:12:20.717+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[2025-10-18T16:12:20.717+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[2025-10-18T16:12:20.718+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[2025-10-18T16:12:20.718+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[2025-10-18T16:12:20.719+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[2025-10-18T16:12:20.720+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:12:20.721+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:12:20.721+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:12:20.724+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:12:20.725+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:12:20.725+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:12:20.726+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:12:20.729+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:12:20.730+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:12:20.788+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:12:20.790+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:12:20.793+0000] {spark_submit.py:492} INFO - ... 37 more
[2025-10-18T16:12:20.794+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:12:20.796+0000] {spark_submit.py:492} INFO - 🧹 Stopping Spark session...
[2025-10-18T16:12:20.798+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-10-18T16:12:20.799+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO SparkUI: Stopped Spark web UI at http://2ef039423968:4040
[2025-10-18T16:12:20.800+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-10-18T16:12:20.801+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO MemoryStore: MemoryStore cleared
[2025-10-18T16:12:20.802+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO BlockManager: BlockManager stopped
[2025-10-18T16:12:20.803+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-10-18T16:12:20.804+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-10-18T16:12:20.805+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:20 INFO SparkContext: Successfully stopped SparkContext
[2025-10-18T16:12:21.392+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:21 INFO ShutdownHookManager: Shutdown hook called
[2025-10-18T16:12:21.398+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:21 INFO ShutdownHookManager: Deleting directory /tmp/spark-8ab83a8f-64bd-44bd-ac79-8950fc307191
[2025-10-18T16:12:21.421+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:21 INFO ShutdownHookManager: Deleting directory /tmp/spark-4627a484-83b6-40c2-9f66-90879bc4b92c/pyspark-6749cf35-acb0-44ff-9de1-bdb68c6617c0
[2025-10-18T16:12:21.517+0000] {spark_submit.py:492} INFO - 25/10/18 16:12:21 INFO ShutdownHookManager: Deleting directory /tmp/spark-4627a484-83b6-40c2-9f66-90879bc4b92c
[2025-10-18T16:12:21.918+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 423, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --name arrow-spark --verbose /opt/***/dags/hdfs_spark_test.py. Error code is: 1.
[2025-10-18T16:12:21.928+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=test_hdfs_spark, task_id=hdfs_spark_test, execution_date=20251018T161124, start_date=20251018T161131, end_date=20251018T161221
[2025-10-18T16:12:22.018+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 98 for task hdfs_spark_test (Cannot execute: spark-submit --master local --name arrow-spark --verbose /opt/***/dags/hdfs_spark_test.py. Error code is: 1.; 3651)
[2025-10-18T16:12:22.090+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-10-18T16:12:22.142+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
